framework/src/modules/reward/module.ts

```diff
@@ -62,9 +62,22 @@ export class RewardModule extends BaseModule {
 		});
 	}
 
-	// eslint-disable-next-line @typescript-eslint/require-await
-	public async afterBlockExecute(_context: BlockAfterExecuteContext): Promise<void> {
-		// eslint-disable-next-line no-console
-		console.log(this._tokenAPI, this._bftAPI, this._randomAPI, this._tokenIDReward);
+	public async afterBlockExecute(context: BlockAfterExecuteContext): Promise<void> {
+		const blockReward = await this.api.getBlockReward(
+			context.getAPIContext(),
+			context.header,
+			context.assets,
+		);
+
+		if (blockReward <= BigInt(0)) {
```

Comment:
`0` reward should be valid, and it is calculating by itself, so no validation is needed?

Reply:
I thought minting 0 reward is useless and redundant, don't you think so? Because if I don't end the control flow there the API will mint 0 reward.

Reply:
On the other hand, 0 reward is valid and not minting is `mint` functions responsibility, so probably I should remove this conditional. Still waiting for your approval.

Reply:
yes, then it should be if `blockReward <= 0` return. if thrown error, block will be rejected

Reply:
done